R/choose_value.R

```diff
@@ -0,0 +1,101 @@
+#' Choose a value from a dataframe
+#'
+#' Returns the value of a given variable from a dataframe for a given time period, and optionally for
+#' a given value in an additional column.
+#'
+#' @param df A dataframe.
+#' @param period_col_name Name of the column containing time periods.
+#' @param period_val Time period to choose a corresponding value for.
+#' @param value_col_name Name of the column containing the variable we want to extract values from.
+#' @param additional_col_name An additional variable name to match a value for when selecting a value.
+#' @param additonal_val Value of the additional variable to be matched when selecting a value.
+#'
+#' @return A numeric value or character object.
+#
+#' @examples
+#' df <- data.frame("yrqtr" = c("2015 Q4", "2015 Q4", "2015 Q4", "2016 Q1", "2016 Q1", "2016 Q1"),
+#' "type" = c("receipts", "disposals", "outstanding", "receipts", "disposals", "outstanding"),
+#' "total" = 1:6
+#' )
+#'
+#' choose_value(df, "yrqtr", "2015 Q4", "total", "type", "outstanding") # returns 3
+#'
+#' @export
+
+choose_value <- function(df, period_col_name, period_val, value_col_name, additional_col_name = NULL, additonal_val = NULL){
+
```

Comment:
Rather than "additional" arguments (which were complicated to use imo), it might be better if the period_col_name and period_val would accept vectors of arguements?